packages/ui/src/components/utils/flex.styled.ts

```diff
@@ -1,57 +1,98 @@
 import { ReactNode } from 'react';
 import { css } from 'styled-components';
+import { MediaQuery } from '../../types';
+import { generateCss } from './generateCss';
+
+type JustifyType =
+  | 'flex-start'
+  | 'center'
+  | 'space-around'
+  | 'space-between'
+  | 'flex-end'
+  | 'evenly';
+
+type AlignType = 'flex-start' | 'center' | 'baseline' | 'flex-end' | 'stretch';
+
+type DirectionType = 'row' | 'column';
+
+type AlignSelfType =
+  | 'auto'
+  | 'stretch'
+  | 'center'
+  | 'start'
+  | 'end'
+  | 'baseline'
+  | 'initial'
+  | 'inherit';
+
+const alignSelfObj: Record<string, string> = {
+  start: 'flex-start',
+  end: 'flex-end',
+};
 
 export interface FlexProps {
   children?: ReactNode;
-  wrapReverse?: boolean;
-  noWrap?: boolean;
-  justify?:
-    | 'flex-start'
-    | 'center'
-    | 'space-around'
-    | 'space-between'
-    | 'flex-end'
-    | 'evenly';
-  align?: 'flex-start' | 'center' | 'baseline' | 'flex-end' | 'stretch';
-  content?:
-    | 'flex-start'
-    | 'flex-end'
-    | 'center'
-    | 'space-between'
-    | 'space-around';
-  direction?: 'row' | 'column';
-  gap?: number;
-  grow?: number;
+  wrapReverse?: MediaQuery<boolean>;
+  noWrap?: MediaQuery<boolean>;
+  justify?: MediaQuery<JustifyType>;
+  align?: MediaQuery<AlignType>;
+  direction?: MediaQuery<DirectionType>;
+  gap?: MediaQuery<number>;
+  grow?: MediaQuery<number>;
+  $alignSelf?: MediaQuery<AlignSelfType>;
 }
 
 const justifyContent = css<FlexProps>`
-  ${props => props.justify && `justify-content: ${props.justify};`}
+  ${props =>
+    props.justify &&
+    generateCss(
+      ['justify-content'],
+      (item: JustifyType) => `${item}`,
+      props.justify,
+    )}
 `;
 
 const align = css<FlexProps>`
-  ${props => props.align && `align-items: ${props.align};`}
+  ${props =>
+    props.align &&
+    generateCss(['align-items'], (item: AlignType) => `${item}`, props.align)}
 `;
 
 const direction = css<FlexProps>`
-  ${props => props.direction && `flex-direction: ${props.direction};`}
+  ${props =>
+    props.direction &&
+    generateCss(
+      ['flex-direction'],
+      (item: DirectionType) => item,
+      props.direction,
+    )}
 `;
 
 const gap = css<FlexProps>`
-  ${props => props.gap !== undefined && `gap: ${props.gap}px;`}
+  ${props =>
+    props.gap && generateCss(['gap'], (item: number) => `${item}px`, props.gap)}
 `;
 const grow = css<FlexProps>`
-  ${props => props.grow !== undefined && `flex-grow: ${props.grow};`}
+  ${props =>
+    props.grow &&
+    generateCss(['flex-grow'], (item: number) => `${item}`, props.grow)}
 `;
 
-const content = css<FlexProps>`
-  ${props => props.content && `align-content: ${props.content};`}
+const $alignSelf = css<FlexProps>`
```

Comment:
Why does this variable start with `$`?

Reply:
_camelCase_ is not allowed in HTML element's props, so to use camelCase we have to put `$` sign, that is a feature in _styled components_ which allows us to hide the props in the rendered HTML elements.